src/components/metadata/listing/index.js

```diff
@@ -0,0 +1,352 @@
+            <TableCell>{unit}</TableCell>
+            <TableCell padding="none" align="right">
+                {avuChildCount}
+            </TableCell>
+            <TableCell padding="none">
+                <Grid
+                    container
+                    spacing={0}
+                    wrap="nowrap"
+                    direction="row"
+                    justify="center"
+                    alignItems="center"
+                >
+                    <Grid item>
+                        <IconButton
+                            id={build(rowID, ids.BUTTONS.EDIT)}
+                            aria-label={t("edit")}
+                            className={classes.button}
+                            onClick={(event) => {
+                                event.stopPropagation();
+                                onRowEdit();
+                            }}
+                        >
+                            {editable ? <ContentEdit /> : <ContentView />}
+                        </IconButton>
+                    </Grid>
+                    {editable && (
+                        <Grid item>
+                            <IconButton
+                                id={build(rowID, ids.BUTTONS.DELETE)}
+                                aria-label={t("delete")}
+                                classes={{ root: classes.deleteIcon }}
+                                onClick={(event) => {
+                                    event.stopPropagation();
+                                    onRowDelete();
+                                }}
+                            >
+                                <ContentRemove />
+                            </IconButton>
+                        </Grid>
+                    )}
+                </Grid>
+            </TableCell>
+        </TableRow>
+    );
+};
+
+const MetadataList = (props) => {
+    const {
+        parentID,
+        editable,
+        loading,
+        fetchError,
+        selectable,
+        onEditAVU,
+        onSelectAVU,
+        onSelectAllClick,
+        avusSelected,
+        rowsInPage,
+        name,
+        form: { values },
+        remove,
+        unshift,
+    } = props;
+
+    const [newAttrCount, setNewAttrCount] = React.useState(1);
+    const [orderBy, setOrderBy] = React.useState(props.orderBy);
+    const [order, setOrder] = React.useState(props.order || "asc");
+
+    const { t } = useTranslation("metadata");
+    const classes = useStyles();
+
+    const onAddAVU = () => {
+        const avus = getIn(values, name) || [];
+
+        let newName,
+            count = newAttrCount;
+
+        const namesMatch = (avu) => avu.attr === newName;
+        do {
+            newName = t("newAttrName", {
+                count,
+            });
+            count++;
+        } while (avus.find(namesMatch));
+
+        setNewAttrCount(count);
+
+        unshift({
+            attr: newName,
+            value: "",
+            unit: "",
+        });
+    };
```

Comment:
It's probably just really late for me to be reviewing PRs, but I'm having trouble understanding what's going on in this `onAddAVU` function.  Maybe a comment would be helpful?

Reply:
👌 Hopefully renaming `newName` to `newAttrName` and the inline comments make it clearer what's going on 😅

Reply:
Yes - this is perfect - thanks!